tests/cases/py_client/lib_test.py

```diff
@@ -144,10 +173,10 @@ def testRestCore(self):
         flag = False
         try:
             self.client.getResource('user/badId')
-        except girder_client.HttpError as e:
-            self.assertEqual(e.status, 400)
-            self.assertEqual(e.method, 'GET')
-            resp = json.loads(e.responseText)
+        except requests.HTTPError as e:
+            self.assertEqual(e.response.status_code, 400)
+            self.assertEqual(e.request.method, 'GET')
```

Comment:
Ah, got it. I glanced at that code, but only the first few lines where `self.response` was explicitly set from a parameter. I don't think this code needs to be complicated any further, unless there are places where the passed `response` doesn't actually contain the `request`.